src/TouchFree.ts

```diff
@@ -39,10 +44,46 @@ const Init = (tfInitParams?: TfInitParams): void => {
     }
 };
 
+const analyticEvents: { [key in AnalyticEventKey]?: () => void } = {};
+
+let sessionEvents: AnalyticSessionEvents = {};
+const defaultAnalyticEvents: AnalyticEventKey[] = ['touchstart', 'touchmove', 'touchend'];
+
+// Function: RegisterAnalyticEvents
+// Registers a given list of event for the TouchFree service to record.
+// If no list of events is provided then the default set of events will be recorded.
+const RegisterAnalyticEvents = (eventsIn: AnalyticEventKey[] = defaultAnalyticEvents) => {
+    eventsIn.forEach((evt) => {
+        const onEvent = () => {
+            const eventCount = sessionEvents[evt];
+            sessionEvents[evt] = eventCount === undefined ? 1 : eventCount + 1;
+        };
+        analyticEvents[evt] = onEvent;
+        document.addEventListener(evt, onEvent, true);
+    });
+};
+
+// Function: UnregisterAnalyticEvents
+// Unregister any registered analytic events.
+// If no list of events is provided then all registered analytic events will be unregistered.
+const UnregisterAnalyticEvents = (eventsIn?: AnalyticEventKey[]) => {
+    const events: AnalyticEventKey[] = eventsIn ? eventsIn : (Object.keys(analyticEvents) as AnalyticEventKey[]);
```

Comment:
:shrug: Could use `eventsIn ??` instead?

Reply:
I avoided this for due to the XT4 but it's used in other places and is much more readable so will change back 🤷